invoke-ai/InvokeAI
# feat(nodes): organise shared node classes/objects

## What type of PR is this? (check all applicable)



## Have you discussed this change with the InvokeAI team?

## Description

[fix(nodes): create app/shared/ module to prevent circular imports](https://github.com/invoke-ai/InvokeAI/pull/5053/commits/f77c0fb06ac22f6aa4bc63f55b4e434f6eb7c470)

We have a number of shared classes, objects, and functions that are used in multiple places. This causes circular import issues.

This commit creates a new `app/shared/` module to hold these shared classes, objects, and functions.

Initially, only `FreeUConfig` and `FieldDescriptions` are moved here. This resolves a circular import issue with custom nodes.

Other shared classes, objects, and functions will be moved here in future commits.

## Related Tickets & Documents



- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

Add a custom node or two to your `$INVOKEAI_ROOT/nodes/` dir and run the app. Should start up.

This node is known to cause the issue: https://github.com/JPPhoto/ideal-size-node/


## Added/updated tests?
